examples/petsc/bpsdmplex.c

```diff
@@ -100,6 +100,45 @@ static int CreateRestrictionPlex(Ceed ceed, CeedInt P, CeedInt ncomp,
   PetscFunctionReturn(0);
 }
 
+// BC functions
+PetscErrorCode BCsMass(PetscInt dim, PetscReal time, const PetscReal x[],
+                       PetscInt vscale, PetscScalar *u, void *ctx) {
+  for (PetscInt i = 0; i < vscale; i++)
+    u[i] = PetscSqrtScalar(PetscSqr(x[0]) + PetscSqr(x[1]) +
+                           PetscSqr(x[2]));
+
+  return 0;
+}
+
+PetscErrorCode BCsDiff(PetscInt dim, PetscReal time, const PetscReal x[],
+                       PetscInt vscale, PetscScalar *u, void *ctx) {
+  #ifndef M_PI
+#define M_PI    3.14159265358979323846
+  #endif
+  const CeedScalar c[3] = { 0, 1., 2. };
+  const CeedScalar k[3] = { 1., 2., 3. };
+
+  for (PetscInt i = 0; i < vscale; i++) 
+    u[i] = sin(M_PI*(c[0] + k[0]*x[0])) *
+           sin(M_PI*(c[1] + k[1]*x[1])) *
+           sin(M_PI*(c[2] + k[2]*x[2]));
+
+  return 0;
+}
+
+// Create BC label
+static PetscErrorCode CreateBCLabel(DM dm, const char name[]) {
+  int ierr;
+  DMLabel label;
+
```

Comment:
Should use `PetscFunctionBeginUser;`

Reply:
Fixed

Reply:
Still not here? It should be used in any function that calls `CHKERRQ`.

Reply:
Ah, I had added it to the BC functions. Now its on this function.